### DISCUSSION TITLE
Clarification on file.create_new behaviour

### DISCUSSION THREAD

**Discussion by qwekjh** [SilentVoid13/Templater#327] [Help]
I tried to make my daily notes open a weekly note if one does not already exist. I wrote this in the daily note's template:
```js
//...
if (!tp.file.exists("name of weekly note")){
let template = tp.file.find_tfile("Weekly (Template)")
await (tp.file.create_new(template,"name of weekly note"))
}
//more commands unrelated to the weekly note...
```
I did not use `open_new` in `create_new` (so it defaults to `false`).

The problem is that every command after the `create_new` was appended to that new (weekly) file and not the previous (daily) file, what am I missing?
 - Does the `tp.file` variable change to the new file created when `create_new` is complete?
 - How should I use `create_new`?
   - Should I only use it as the last command of a file?
- If I remove the `await`, the commands are appended correctly, but I suspect it is just a race condition

**Comment by welpdx** [comment]
Hey @qwekjh, this is a great question.

According to the [documentation](https://silentvoid13.github.io/Templater/docs/internal-variables-functions/internal-modules/file-module#:~:text=folder%3F%3A%20tfolder), `create_new()` accepts 4 parameters, the last of which is a `TFolder` object that specifies the location of this new note. Because you did not provide that to the `create_new()` function, it defaulted to the folder of the note that the script was running in.

I unfortunately can not find an API function to get a `TFolder` object from path. But for your use case, maybe you can just use the `create()` [function from the obisidian api](https://github.com/obsidianmd/obsidian-api/blob/bde556afa033e909ebfb9fcee8f5ef288276f78f/obsidian.d.ts#L2996) instead. 

```
//...
let name = "name of weekly note"
if (!tp.file.exists(name)){
let template = tp.file.find_tfile("Weekly (Template)")
let templaterContent = await this.app.vault.cachedRead(template);
await this.app.vault.create("/foo/bar/" + name +".md", templaterContent)
}
//more commands unrelated to the weekly note...
```

The above script should do what you wanted to do: 
If file does not exist, 
1. find Weekly Template, 
2. extract the Template's contents, 
3. creates a new file in a certain path with the specified `name`, and 
4. The Templater Plugin should also automatically run the scripts in the Template's contents in the new file shortly after it's creation.  


### demo
![gif](https://i.imgur.com/lDAGFA0.gif)


Hope that helps. 
Welp

**Reply by AB1908** [reply]
`app.vault.getAbstractFileByPath` seems to return a `TFolder` if you pass it a folder's path. Can also be obtained from `TFile`'s `parent` attribute.

**Reply by AB1908** [reply]
`create_new` also defaults to Obsidian's setting for default new note creation IIRC.

**Comment by chuckedfromspace** [comment]
I may want to achieve something else and the method suggested by @welpdx didn't quite work for me. The new file ended up just a simple copy of the template while the tp functions were not executed. I want to use a template to create a file in a specific folder. I was able to do it with the `tp.file.create_new()` function (using the same example):

```
let name = "name of weekly note"
let path = "/foo/"
let template = tp.file.find_tfile("Weekly (Template)")
await tp.file.create_new(template, path+name)
```

**Reply by AB1908** [reply]
If you want to resolve templates, can use `tp.file.include`.
